components/nifosg/nifloader.cpp

```diff
@@ -1626,6 +1626,7 @@ namespace NifOsg
                                                                                    getBlendMode((alphaprop->flags>>5)&0xf)));
                         blendFunc = shareAttribute(blendFunc);
                         stateset->setAttributeAndModes(blendFunc, osg::StateAttribute::ON);
+                        stateset->setAttributeAndModes(new osg::Depth(osg::Depth::LESS, 0.0, 1.0, false));
```

Comment:
As has been mentioned before, the nif files set the depth parameters themselves. Unless you can verify that vanilla does it, it's not up to the engine to override the depth settings the model says to use (including the defaults), even if they're not ideal.

Reply:
please I opened several pr in order to discuss of various mod
use this one https://github.com/OpenMW/openmw/pull/2423  to talk about blendfunc/deptmask

Reply:
and as i repeat to everyone: you MUST disable depthwrite for proper alpha blending...so i don't care what you say unless you have a proper answer to the problem of transparent occlusion

Reply:
I could remove this code by simply remove particle system from TRANSPARENT_BIN but again if we want correct blending of particles with other blended it's mandatory

Reply:
> i don't care what you say unless you have a proper answer

The answer is, the engine doesn't second-guess what the game data tells it to do. Unless you can verify the vanilla engine does it too, it will be up to modders to create fixed models, not for the engine to alter models based on what it thinks it should do.

Reply:
you're wrong : the engine do want it want with game data
it's the principle of a blackbox
if someone have apitrace of a lantern in morrowing, i would really be surprised if they write depth for the glass as it prevent the drawing of the particlesystem

Reply:
No. The engine does what the data says. You don't fix broken data on the engine side. Engines meant for general use are transparent by nature, not the black box you're talking about.

Reply:
I've made 3 engines so don't tell me how magic works

Reply:
and please discuss this problem in the dedicated pr #2423

Reply:
...or on discord...

Reply:
![alphafix](https://user-images.githubusercontent.com/4062709/61089786-8b165380-a43c-11e9-8eaf-1d232ff808ca.png)

Reply:
I remove the merge to make u happy

Reply:
now lanterns  and foutain are broken (under certain view angle)
![Capture d’écran de 2019-07-12 21-11-59](https://user-images.githubusercontent.com/4062709/61152924-041fb480-a4ea-11e9-9713-91ae99d3d8f9.png)
![Capture d’écran de 2019-07-12 21-13-31](https://user-images.githubusercontent.com/4062709/61152955-1994de80-a4ea-11e9-96c4-6cb26d25457d.png)

Reply:
Someone tell me a cell with a dodgy lantern and I'll have a look.

Reply:
https://github.com/OpenMW/openmw/pull/2441/commits/998900566e42bfae8e353e1856eeabaa122665f0 fix the problem